cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh

```diff
@@ -150,6 +150,25 @@ function wait_pods_ready {
     done
 }
 
+function wait_pods_ready_by_label {
+    local -r tries=10
+    local -r namespace=$1
+    local -r label=$2
+
+    ITERATION=0
+    while [ $ITERATION -lt $tries ] && [ ! "$(_kubectl wait pods -n $namespace -l $label --for condition=Ready --timeout 1m)" ]; do
```

Comment:
All other polling functions in this file make use of `retry` for example [wait_for_k8s_object](https://github.com/kubevirt/kubevirtci/pull/481/files#diff-499cff045d8a875e8b39f4a9fd59354eb99e8f8d91a95ebc70fbf42d2d5b5beaR84) 
this was the idea of presenting that function and when we added it iirc you asked for it on your review
Why you dont make use of it here?

Reply:
Its true that i asked it, but not for "wait pods", just for the other things.
I wanted to preserve the elegant use of "wait pods" logic here.
Every thing can be done in several methods, and i did as close as `wait_pods_ready` which already exists,
but preserved the "wait pods".

Reply:
Please see that `retry` doesn't suit the use i need according the comment:

# This function gets a command string and invoke it
# until the command returns an empty string or until timeout

If it returns an empty string i do need to continue.
It shows that we should keep it simple when we can, and stick to existing standard commands (such as "wait").

Reply:
@oshoval I do not think you answered the question presented here.
I did not remember that `retry` function, which seems a total fit here.

You have re-implemented a retry with a `while`. Please try to use the existing `retry` function or explain why it does not fit here technically.

Reply:
At the same time, we should not have multiple "retry" implementations, that is wrong.
Try to see if you can adjust one or the other to fit here.

Reply:
Please see retry docs
```
# This function gets a command string and invoke it
# until the command returns an empty string or until timeout
```
The function doesn't fit naturally, unless the command will be (added the `|| echo error` in order for it to fit)
`_kubectl wait pods -n $namespace -l $label --for condition=Ready --timeout 1m || echo error`

Moreoever we already have `wait_pods_ready` which use a while,
If you still think i should use `retry` i will do that, but i think it doesn't suit here.

Reply:
You should either generalize `retry` better or adjust your function to fit it.
The comment about not creating another "retry" method is valid.